c2rust-analyze/src/rewrite/expr/mir_op.rs

```diff
@@ -697,6 +712,13 @@ where
                     (self.emit)(RewriteKind::UnsafeCastRawToRef { mutbl: true });
                     Some(Ownership::Mut)
                 }
+                Ownership::Cell if !early => {
+                    let printer = FmtPrinter::new(self.tcx, Namespace::TypeNS);
+                    let ty = to.pointee_ty.print(printer).unwrap().into_buffer();
+                    (self.emit)(RewriteKind::CastRawMutToCellPtr { ty });
+                    (self.emit)(RewriteKind::MutToImm);
```

Comment:
This should be an `UnsafeCastRawToRef { mutbl: false }` instead.  `MutToImm` is for `&mut T -> &T`, which is not what we have here (though it happens to generate code that works for this case).

Reply:
74d5f314